packages/core/src/service/services/product-variant.service.ts

```diff
@@ -723,7 +723,7 @@ export class ProductVariantService {
                 const variantOptionIds = this.sortJoin(variant.options, ',', 'id');
                 if (variantOptionIds === inputOptionIds) {
                     throw new UserInputError('error.product-variant-options-combination-already-exists', {
-                        optionNames: this.sortJoin(variant.options, ', ', 'code'),
+                        variantName: variant.name,
```

Comment:
Does variant always have name?

Reply:
Yes, but it is not always available. Since the "name" is translatable, it actually exists in a different table (product_translation), and is "hydrated" into the ProductVariant entity using the `translateDeep()` helper function. In this case, I think the name will be `undefined` since I cannot see anywhere that `translateDeep()` is being applied.

Reply:
I've added but not tested it in another commit. Is this the way to use translateDeep and access the value?

Reply:
From the failing test, it seems that the translateDeep does indeed work! That just means you need to update this e2e test: https://github.com/vendure-ecommerce/vendure/blob/master/packages/core/e2e/product.e2e-spec.ts#L1304

in order to test for the updated error message.